app/models/conversation.rb

```diff
@@ -6,7 +6,7 @@ class Conversation < ApplicationRecord
   has_many :runs, dependent: :destroy
   has_many :steps, dependent: :destroy
 
-  after_create_commit :set_title_async, if: -> { title.blank? }
+  after_create_commit :set_title_async, if: -> { title.blank? && messages.count >= 1 }
```

Comment:
I struggled a bit with the trigger. Initially, after_create_commit + title.blank? seemed right. But `Message.create` creates a conversation a split second before a Message so if the job runs too fast it could run before the message has been created.

Initially, I tried changing this to `after_update_commit` since the creation of a message updates the conversation timestamp and then I added `messages.count >= 1`. But we update conversation A TON because every single update to a message fragment while it's streaming in. I actually wonder if we need to make that more efficient.

So I changed this back to after_create_commit and I added a 2 second delay (down in set_title_async).

But... there's a possibility it runs once and still fails to find any messages. Maybe I need to add a delayed retry.